internal/provision/profiles.go

```diff
@@ -65,38 +56,28 @@ func LoadProfiles(path string, dic *di.Container) errors.EdgeX {
 			fullPath := filepath.Join(absPath, fName)
 			yamlFile, err := ioutil.ReadFile(fullPath)
 			if err != nil {
-				lc.Errorf("failed to read %s: %v", fullPath, err)
+				lc.Errorf("Failed to read %s: %v", fullPath, err)
 				continue
 			}
 
 			err = yaml.Unmarshal(yamlFile, &profile)
 			if err != nil {
-				lc.Errorf("filed to decode profile %s: %v", fullPath, err)
-				continue
-			}
-			// if profile already exists in metadata, skip it
-			if p, ok := profileMap[profile.Name]; ok {
-				_ = cache.Profiles().Add(p)
+				lc.Errorf("Failed to decode profile %s: %v", fullPath, err)
 				continue
 			}
 
-			// add profile to metadata
-			ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.NewString())
-			_, err = dpc.AddByYaml(ctx, fullPath)
-			if err != nil {
-				lc.Errorf("failed to add profile %s to metadata: %v", fullPath, err)
-				continue
+			_, err = dpc.DeviceProfileByName(context.Background(), profile.Name)
```

Comment:
check the existence from cache as you do for "Device"

Reply:
profile is added to cache only if a device is using it. I think  it's more consistent here checking what's on metadata ?

Reply:
you are right